assets/js/utils/style-attributes-utils.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable @wordpress/no-unsafe-wp-apis */
+/**
+ * External dependencies
+ */
+import { __experimentalUseColorProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { isFeaturePluginBuild } from '../settings/blocks/feature-flags';
+import { isString, isObject } from '../types/type-guards';
+
+type WithClass = {
+	className: string;
+};
+
+type WithStyle = {
+	style: Record< string, unknown >;
+};
+
+const parseStyle = ( style: unknown ): Record< string, unknown > => {
+	if ( isString( style ) ) {
+		return JSON.parse( style ) || {};
+	}
+
+	if ( isObject( style ) ) {
+		return style;
+	}
+
+	return {};
+};
+
+const getSpacingStyleInline = (
+	key: string,
+	values: {
+		top: string | null;
+		right: string | null;
+		bottom: string | null;
+		left: string | null;
+	}
+) => {
+	return {
+		...( isString( values.top ) && { [ `${ key }Top` ]: values.top } ),
+		...( isString( values.right ) && {
+			[ `${ key }Right` ]: values.right,
+		} ),
+		...( isString( values.bottom ) && {
+			[ `${ key }Bottom` ]: values.bottom,
+		} ),
+		...( isString( values.left ) && { [ `${ key }Left` ]: values.left } ),
```

Comment:
Can we use key: value here then remove the null properties? We can even return style object with null properties because they will be omitted during render. Using spread here is quite hard to read IMO.